Content.Client/Doors/DoorSystem.Airlock.cs

```diff
@@ -0,0 +1,152 @@
+            state = DoorState.Closed;
+
+        if (_appearanceSystem.TryGetData<bool>(airlock, DoorVisuals.Powered, out var powered, args.Component) && powered)
+        {
+            boltedVisible =
+                _appearanceSystem.TryGetData<bool>(airlock, DoorVisuals.BoltLights, out var lights, args.Component)
+                && lights && state is DoorState.Closed or DoorState.WeldedClosed;
+
+            emergencyLightsVisible =
+                _appearanceSystem.TryGetData<bool>(airlock,
+                    DoorVisuals.EmergencyLights,
+                    out var eaLights,
+                    args.Component) && eaLights;
+            unlitVisible =
+                (state is DoorState.AttemptingCloseBySelf
+                     or DoorState.AttemptingOpenBySelf
+                     or DoorState.AttemptingCloseByPrying
+                     or DoorState.AttemptingOpenByPrying
+                     or DoorState.Denying
+                 || state == DoorState.Open && airlock.Comp.OpenUnlitVisible
+                 || _appearanceSystem.TryGetData<bool>(airlock,
+                     DoorVisuals.ClosedLights,
+                     out var closedLights,
+                     args.Component) && closedLights)
+                && !boltedVisible && !emergencyLightsVisible;
+        }
+
+        args.Sprite.LayerSetVisible(DoorVisualLayers.BaseUnlit, unlitVisible);
+        args.Sprite.LayerSetVisible(DoorVisualLayers.BaseBolted, boltedVisible);
+
+        if (airlock.Comp.EmergencyAccessLayer)
+        {
+            args.Sprite.LayerSetVisible(
+                DoorVisualLayers.BaseEmergencyAccess,
+                emergencyLightsVisible
+                && state != DoorState.Open
+                && state != DoorState.AttemptingOpenBySelf
+                && state != DoorState.AttemptingCloseBySelf
+                && !boltedVisible
+            );
+        }
+
+        switch (state)
+        {
+            case DoorState.Open:
+                args.Sprite.LayerSetState(DoorVisualLayers.BaseUnlit, airlock.Comp.ClosingSpriteState);
+                args.Sprite.LayerSetAnimationTime(DoorVisualLayers.BaseUnlit, 0);
+
+                break;
+            case DoorState.Closed:
+                args.Sprite.LayerSetState(DoorVisualLayers.BaseUnlit, airlock.Comp.OpeningSpriteState);
+                args.Sprite.LayerSetAnimationTime(DoorVisualLayers.BaseUnlit, 0);
+
+                break;
+            case DoorState.AttemptingCloseBySelf:
+            case DoorState.AttemptingCloseByPrying:
+            case DoorState.Closing:
+            case DoorState.AttemptingOpenBySelf:
+            case DoorState.AttemptingOpenByPrying:
+            case DoorState.Opening:
+            case DoorState.WeldedClosed:
+            case DoorState.Denying:
+            case DoorState.Emagging:
+            default:
+                break;
+        }
```

Comment:
```suggestion

        if (state == DoorState.Open)
        {
            args.Sprite.LayerSetState(DoorVisualLayers.BaseUnlit, airlock.Comp.ClosingSpriteState);
            args.Sprite.LayerSetAnimationTime(DoorVisualLayers.BaseUnlit, 0);
        }
        else if (state == DoorState.Closed)
        {
            args.Sprite.LayerSetState(DoorVisualLayers.BaseUnlit, airlock.Comp.OpeningSpriteState);
            args.Sprite.LayerSetAnimationTime(DoorVisualLayers.BaseUnlit, 0);
        }
```
imo a switch with more empty cases than filled ones should just be an `if`/`else`.

Reply:
I'll probably cull this to the two filled cases and default.